modules/ocl-loader/src/main/java/org/eclipse/xpanse/modules/ocl/loader/data/models/Ocl.java

```diff
@@ -31,15 +33,34 @@
 public class Ocl {
 
     private static ObjectMapper theMapper = new ObjectMapper();
+
     @NotBlank
     @Schema(description = "Name of the service to be managed by xpanse")
     private String name;
-    private String category;
+
+    @Valid
+    @NotNull
+    @Schema(description = "The category of the managed service")
+    private Category category;
+
+    @NotNull
+    @Schema(description = "The namespace of the managed service")
     private String namespace;
-    private Map<String, Object> properties;
+
+    @Valid
+    @Schema(description = "The billing of the managed service")
     private Billing billing;
+
+    @Valid
+    @Schema(description = "The compute resources for the managed service")
     private Compute compute;
+
+    @Valid
+    @Schema(description = "The network resources for the managed service")
     private Network network;
+
+    @Valid
+    @Schema(description = "The storage resources for the managed service")
     private List<Storage> storages;
```

Comment:
for all the lists in our data models. which are not mandatory.. we must add a min size of 1. 
With that the validation would allow list as null but if user wishes to give a list, then empty list cannot be sent.